lean4-infoview/src/infoview/goalLocation.tsx

```diff
@@ -76,66 +76,125 @@ export interface Locations {
 
 export const LocationsContext = React.createContext<Locations | undefined>(undefined)
 
-type SelectableLocationProps = React.PropsWithoutRef<
-    React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>
-> & {
-    locs?: Locations
-    loc?: GoalsLocation
-    alwaysHighlight: boolean
-    setHoverState?: React.Dispatch<React.SetStateAction<HoverState>>
+export type HoverSettings = { highlightOnHover: true } | { highlightOnHover: false }
```

Comment:
It's not, but I felt that the API would be more consistent this way and easier to extend in case we do eventually add new options for these cases. Happy to use either.

Reply:
Since the types are literally the same, it should make no difference to future extensibility I think. One is just shorter.